core: rtw_ieee80211.c: define CONFIG_GET_MAC_ADDRESS_FROM_FILE

RadxaMitchell opened the pull request:
Support for reading MAC addresses from files by adding CONFIG_GET_MAC_ADDRESS_FROM_FILE

RadxaYuntian commented on debian/patches/get-mac-address-from-file.patch:
```diff
@@ -0,0 +1,46 @@
+  *
+  ******************************************************************************/
+ #define _IEEE80211_C
++#define CONFIG_GET_MAC_ADDRESS_FROM_FILE
```
不要直接在文件里面定义，在[Makefile](https://github.com/lwfinger/rtl8723ds/blob/52e593e8c889b68ba58bd51cbdbcad7fe71362e4/Makefile#L649)里面用`EXTRA_CFLAGS += -DCONFIG_GET_MAC_ADDRESS_FROM_FILE`定义

RadxaYuntian commented on debian/patches/get-mac-address-from-file.patch:
```diff
@@ -0,0 +1,46 @@
+        int i;
+        struct file *fp = NULL;
+-       mm_segment_t oldfs;
++       //mm_segment_t oldfs;
```
这个变量到底有没有用到？如果只有CONFIG_PLATFORM_INTEL_BYT用到了那就加上ifdef隔离出去

RadxaYuntian commented on debian/patches/get-mac-address-from-file.patch:
```diff
@@ -0,0 +1,46 @@
++       //mm_segment_t oldfs;
+        unsigned char c_mac[MAC_ADDRESS_LEN];
+-       char fname[] = "/config/wifi/mac.txt";
++       char fname[] = "/etc/wifi/rtl8723ds_mac.txt";
```
直接保存在/etc目录里面，别的子目录有可能不存在

RadxaYuntian reviewed (changes requested):
另外我没有看到如果这个文件不存在又没有读取到efuse时的生成随机mac地址以及保存的代码

RadxaMitchell commented:
> 另外我没有看到如果这个文件不存在又没有读取到efuse时的生成随机mac地址以及保存的代码

这个要用户在 /etc 下添加文件添加生成的 mac 地址

RadxaYuntian reviewed (changes requested):
两个补丁合一起

RadxaYuntian commented:
另外应该要加上保存随机生成的mac地址功能，用户在使用时不应该需要自己去手动添加这个文件才能正常工作

Commit:
debian: patches: Add patch to get wifi mac address from file

What this patch does is to save the randomly generated MAC address to a file,
and read the written MAC address from the file right after the next boot.

Signed-off-by: Chuang Ma <machaung@radxa.com>

RadxaYuntian commented:
有自己生成deb包尝试安装么？能在产品上编译通过并且mac地址工作正常么？

RadxaMitchell commented:
> 有自己生成deb包尝试安装么？能在产品上编译通过并且mac地址工作正常么？

试了，我用生成的 deb 包在 rock-pi-s 上安装后，可以编译通过，mac 地址也工作正常
